src/main/java/com/ase/restservice/ApplicationSecurity.java

```diff
@@ -0,0 +1,137 @@
+package com.ase.restservice;
+
+import com.ase.restservice.jwt.JwtTokenFilter;
+import com.ase.restservice.model.Account;
+import com.ase.restservice.repository.AccountRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+import javax.annotation.Resource;
+import javax.servlet.http.HttpServletResponse;
+
+
+@Configuration
+public class ApplicationSecurity {
+
+  @Resource
+  private UserDetailsService userDetailsService;
+
+  @Autowired
+  private AccountRepository accountRepo;
+  @Autowired
+  private JwtTokenFilter jwtTokenFilter;
+
+  /**
+   * This method checks whether the account asked for
+   * by the client is authorized to the client or not.
+   * @param authentication authenticated client object
+   * @param accountId accountId that client wants to open
+   * @return whether client is allowed to access account or not
+   */
+  public boolean checkAccountId(Authentication authentication, String accountId) {
+    // need to check which client this account has
+    String clientName = authentication.getName();
+    Account account =  accountRepo.findAccountByAccountId(accountId).orElseThrow(() ->
+            new UsernameNotFoundException("Account Not Found with username: " + accountId));
```

Comment:
Can we throw our AccountNotFoundException here?

Reply:
It appears that we cannot change it because UnsernameNotFoundException is an exception made for Spring Security. I am looking into finding or implementing a different one that suits our needs. However AccountNotFoundException cannot be used here.